src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/HelpersWindows.cs

```diff
@@ -532,23 +532,21 @@ private static unsafe void AddCryptAttribute(CryptographicAttributeObjectCollect
             collection.Add(new CryptographicAttributeObject(oid, attributeCollection));
         }
 
-        private static void GetParameters(CRYPT_ALGORITHM_IDENTIFIER cryptAlgorithmIdentifer, ref byte[] parameters)
+        private static void GetParameters(CRYPT_ALGORITHM_IDENTIFIER cryptAlgorithmIdentifer, int cbSize, ref byte[] parameters)
         {
-            if (cryptAlgorithmIdentifer.Parameters.cbData > 0)
+            if (cryptAlgorithmIdentifer.Parameters.cbData > 0 && cbSize > 0)
```

Comment:
When is it not zero?  Why is it passed by the caller?

Reply:
I'm not sure how to correctly get it inside the method so I relied on the values which NET Framework returns for particular algorithm oids.